atomic_reactor/config.py

```diff
@@ -0,0 +1,567 @@
+                                                                organization)
+
+    def _get_cluster_configuration(self):
+        all_cluster_configs = {}
+        for platform, clusters in self.clusters.items():
+            cluster_configs = [ClusterConfig(priority=priority, **cluster)
+                               for priority, cluster in enumerate(clusters)]
+            all_cluster_configs[platform] = cluster_configs
+        return all_cluster_configs
+
+    def get_enabled_clusters_for_platform(self, platform):
+        cluster_configs = self._get_cluster_configuration()
+        if platform not in cluster_configs:
+            return []
+        return [conf for conf in cluster_configs[platform] if conf.enabled]
+
+    def cluster_defined_for_platform(self, platform):
+        cluster_configs = self._get_cluster_configuration()
+        return platform in cluster_configs
+
+    def get_odcs_config(self):
+        """
+        Return an odcs config object created from the odcs config configured in
+        reactor config
+
+        :return: the object of ODCSConfig. If there is no odcs configured in
+            reactor config, None is returned.
+        :rtype: :class:`ODCSConfig` or None
+        """
+        if self.odcs:
+            return ODCSConfig(
+                signing_intents=self.odcs['signing_intents'],
+                default_signing_intent=self.odcs['default_signing_intent']
+            )
+
+    def is_default(self):
+        return self.conf == self.DEFAULT_CONFIG
+
+    def _get_value(self, name, fallback=NO_FALLBACK):
+        try:
+            # make a deep copy to prevent plugins from changing the value for other plugins
+            value = deepcopy(self.conf[name])
+            return value
+        except KeyError:
+            if fallback != NO_FALLBACK:
+                return fallback
+            raise
+
+    @property
+    def koji(self):
+        koji_map = self._get_value(ReactorConfigKeys.KOJI_KEY)
+
+        if 'auth' in koji_map:
+            krb_principal = koji_map['auth'].get('krb_principal')
+            krb_keytab = koji_map['auth'].get('krb_keytab_path')
+            if bool(krb_principal) != bool(krb_keytab):
+                raise RuntimeError("specify both koji_principal and koji_keytab or neither")
+
+        return koji_map
+
+    @property
+    def koji_session(self):
+        from atomic_reactor.utils.koji import create_koji_session
+
+        auth_info = {
+            "proxyuser": self.koji['auth'].get('proxyuser'),
+            "ssl_certs_dir": self.koji['auth'].get('ssl_certs_dir'),
+            "krb_principal": self.koji['auth'].get('krb_principal'),
+            "krb_keytab": self.koji['auth'].get('krb_keytab_path')
+        }
+
+        use_fast_upload = self.koji.get('use_fast_upload', True)
+
+        return create_koji_session(self.koji['hub_url'], auth_info, use_fast_upload)
+
+    @property
+    def koji_path_info(self):
+        from koji import PathInfo
```

Comment:
Why import is here?

Reply:
it was like that in reactor_config_map plugin, but I can move all those imports to the top if you want